### PR TITLE
chore(release): update changelog for 2021-09 release

### PR DESCRIPTION
**Pull request by dbolduc**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7245)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -64,6 +102,12 @@
   removed. If your code previously relied on `sor.status() = new_status` you
   should change it to `sor = new_status`. ([#7150](https://github.com/googleapis/google-cloud-cpp/pull/7150))
 
+**OTHER CHANGES**:
+
+* feat(common): add GrpcChannelArgumentsNativeOption ([#7194](https://github.com/googleapis/google-cloud-cpp/pull/7194))
+* feat(common): capture thread that creates log record ([#7119](https://github.com/googleapis/google-cloud-cpp/pull/7119))
+
+
```
Comment (devbww) [inline CHANGELOG.md:110]: Nitty nit: Delete extra blank line.

File: CHANGELOG.md [hunk 2]
```diff
@@ -52,9 +52,47 @@
...
+* feat(pubsub): support Topic retention duration ([#7196](https://github.com/googleapis/google-cloud-cpp/pull/7196))
+
+### [Storage](https://github.com/googleapis/google-cloud-cpp/blob/main/google/cloud/storage/README.md)
+
+* fix: release CurlRequest handles on error ([#7244](https://github.com/googleapis/google-cloud-cpp/pull/7244))
+* feat(storage): add additional examples ([#7224](https://github.com/googleapis/google-cloud-cpp/pull/7224))
+* feat(storage): aggregate upload benchmark ([#7190](https://github.com/googleapis/google-cloud-cpp/pull/7190))
+* fix(GCS+gRPC): quickstart build with Bazel+Windows ([#7147](https://github.com/googleapis/google-cloud-cpp/pull/7147))
+* refactor(storage): isolate header parsing to REST ([#7137](https://github.com/googleapis/google-cloud-cpp/pull/7137))
```
Comment (coryan) [inline CHANGELOG.md:81]: Typically we do not include `refactor` information in the CHANGELOG.

**Review by coryan — APPROVED** [review]
One minor nit.

File: CHANGELOG.md (hunk 2 above)
Comment (devjgm) [reply CHANGELOG.md:81]: We could update https://github.com/googleapis/google-cloud-cpp/blob/be1513b5067b11a94f3aa8b71edcd581e0927d62/release/changes.sh#L51 to exclude `refactor` by default.

**Review by devjgm — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (dbolduc) [reply CHANGELOG.md:81]: I removed others, but wasn't sure about this one. I will remove it.

File: CHANGELOG.md (hunk 1 above)
Comment (dbolduc) [reply CHANGELOG.md]: Fixed

**Pull request merged**